### PR TITLE
Apply hlint sugestions to alonzo and shelley

### PR DESCRIPTION
**Pull request by lehins**
This PR applies most suggestions from hlint, which mostly just:

* redundant parenthesis
* unused extensions
* eta reduction
* single argument data types changed to newtypes

Feel free to just mention on any line if the change needs to be reversed, I am not going to push back.

There are a few minor, but notable improvements from me:

* `explain_plutus_failure` -> `explainPlutusFailure` rename with deprecation (although quick search didn't show any usage from outside)
* pretty printing module got most fixups, in particular instances got formatted to be more consistent
* some long lines that I found which didn't fit on a screen without wrapping around got reformatted.
* Removed usage of deprecated `"isValidating"` in a couple of places
* haddock improvements

**Important** - none of the changes in this PR affects the semantics of any functions (there might be a bit more strictness due to  newtypes, but that should be a desired improvement IMHO)

I'll be happy to apply this sort of cleanup to the rest of the repository once this PR is merged.

### CODE REVIEW

File: alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxos.hs [hunk 1]
```diff
@@ -232,14 +228,14 @@ scriptsNotValidateTransition = do
...
-      case (evalScripts @era tx sLst) of
-        Passes -> False ?!## ValidationTagMismatch (getField @"isValidating" tx) PassedUnexpectedly
+      case evalScripts @era tx sLst of
+        Passes -> False ?!## ValidationTagMismatch (getField @"isValid" tx) PassedUnexpectedly
         Fails _sss -> pure ()
     Left info -> failBecause (CollectErrors info)
   pure $
     us
       { _utxo = eval (getField @"collateral" txb ⋪ utxo),
-        _fees = fees <> Val.coin (balance @era (eval (getField @"collateral" txb ◁ utxo)))
```
Comment (nc6) [inline alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxos.hs]: I find the qualified import slightly more helpful here, but ymmv.

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/EpochBoundary.hs [hunk 2]
```diff
@@ -210,12 +207,8 @@ instance
...
   FromCBOR (SnapShot crypto)
   where
-  fromCBOR = do
-    decodeRecordNamed "SnapShot" (const 3) $ do
-      s <- fromCBOR
-      d <- fromCBOR
-      p <- fromCBOR
-      pure $ SnapShot s d p
+  fromCBOR =
+    decodeRecordNamed "SnapShot" (const 3) $ SnapShot <$> fromCBOR <*> fromCBOR <*> fromCBOR
```
Comment (nc6) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/EpochBoundary.hs]: Slightly long line

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs [hunk 3]
```diff
@@ -281,11 +273,8 @@ instance CC.Crypto crypto => ToCBOR (FutureGenDeleg crypto) where
     encodeListLen 2 <> toCBOR a <> toCBOR b
 
 instance CC.Crypto crypto => FromCBOR (FutureGenDeleg crypto) where
-  fromCBOR = do
-    decodeRecordNamed "FutureGenDeleg" (const 2) $ do
-      a <- fromCBOR
-      b <- fromCBOR
-      pure $ FutureGenDeleg a b
+  fromCBOR =
+    decodeRecordNamed "FutureGenDeleg" (const 2) $ FutureGenDeleg <$> fromCBOR <*> fromCBOR
```
Comment (nc6) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs]: Ditto

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs [hunk 4]
```diff
@@ -373,15 +362,15 @@ instance
...
-      pure $ DState rw dlg p fgs gs ir
+  fromCBOR =
+    decodeRecordNamed "DState" (const 6) $
+      DState
+        <$> fromCBOR
+        <*> fromCBOR
+        <*> fromCBOR
+        <*> fromCBOR
+        <*> fromCBOR
+        <*> fromCBOR
```
Comment (nc6) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs]: I think the named version is slightly easier to read here

**Review by nc6 — APPROVED** [review]
Great effort, long overdue! I left some very minor comments.

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs (hunk 4 above)
Comment (lehins) [reply shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs]: Reverted

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs (hunk 3 above)
Comment (lehins) [reply shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs]: Done

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/EpochBoundary.hs (hunk 2 above)
Comment (lehins) [reply shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/EpochBoundary.hs]: Split in two lines. Done

File: alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxos.hs (hunk 1 above)
Comment (lehins) [reply alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxos.hs]: Reverted. Done

**Review by JaredCorduan — APPROVED** [review]
fantastic, thank you!

**Pull request merged**